lib/improver/tests/lapse_rate/Test_LapseRate.py

```diff
@@ -0,0 +1,367 @@
+        self.assertIsInstance(result, Cube)
+
+    def test_fails_if_data_is_not_cube(self):
+        """Test code raises a Type Error if input cubes are not cubes."""
+        incorrect_input = 50.0
+
+        msg = 'Temperature input is not a cube, but {0}'.format(
+            type(incorrect_input))
+        with self.assertRaisesRegexp(TypeError, msg):
+            LapseRate().process(incorrect_input, self.orography,
+                                self.land_sea_mask)
+
+        msg = 'Orography input is not a cube, but {0}'.format(
+            type(incorrect_input))
+        with self.assertRaisesRegexp(TypeError, msg):
+            LapseRate().process(self.temperature, incorrect_input,
+                                self.land_sea_mask)
+
+        msg = 'Land/Sea mask input is not a cube, but {0}'.format(
+            type(incorrect_input))
+        with self.assertRaisesRegexp(TypeError, msg):
+            LapseRate().process(self.temperature, self.orography,
+                                incorrect_input)
+
+    def test_fails_if_cube_wrong_units(self):
+        """Test code raises a Value Error if input cubes are the
+           wrong units."""
+
+        # Copies cube and converts to wrong units.
+        incorrect_temp = self.temperature.copy()
+        incorrect_temp.convert_units('celsius')
+        incorrect_orog = self.orography.copy()
+        incorrect_orog.convert_units('feet')
+
+        msg = 'Temperature cube units are {}, must be Kelvin!'.format(
+           incorrect_temp.units)
+        with self.assertRaisesRegexp(ValueError, msg):
+            LapseRate().process(incorrect_temp, self.orography,
+                                self.land_sea_mask)
+
+        msg = 'Orography cube units are {}, must be metres!'.format(
+           incorrect_orog.units)
+        with self.assertRaisesRegexp(ValueError, msg):
+            LapseRate().process(self.temperature, incorrect_orog,
+                                self.land_sea_mask)
+
+    def test_return_single_precision(self):
+        """Test that the function returns cube of float32."""
+        result = LapseRate().process(self.temperature, self.orography,
+                                     self.land_sea_mask)
+        self.assertEqual(result.dtype, np.float32)
+
+    def test_constant_temp_orog(self):
+        """Test that the function returns expected 0.01980198 values where the
+           temperature and orography fields are constant values.
+           This value is because, where the data is such that the gradient
+           value is meaningless, then numpy.linalg.lstsq will calculate
+           the gradient assuming that the intercept is zero.
+           The mathematical validity of this approach is unclear. The FORTRAN
+           code used in the operational system has statements that catch
+           odd gradient values and subistuite them with DALR. This will
+           require further investigation.
+        """
```

Comment:
This doesn't need to be all in this docstring.  Please move discussion of the validity of the current method to the relevant JIRA ticket.